### ISSUE TITLE
iOS 7.1.0: Crash when editing start or end times of old entry

### ISSUE THREAD

**Issue by alfonsogarciacaro** [toggl/mobile#1591]
When a user tries to edit the start or stop times of ac entry made with an old version of the app, the app crashes. Note this doesn't happen with new entries created with v7.1.0.

The exception can be reproduced with the iOS simulator, occurs [here](https://github.com/toggl/mobile/blob/master/Ross/ViewControllers/EditTimeEntryViewController.cs#L370) and displays the following output:

```
Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: UITableView dataSource is not set
Native stack trace:
	0   CoreFoundation                      0x03d9e494 __exceptionPreprocess + 180
	1   libobjc.A.dylib                     0x04415e02 objc_exception_throw + 50
	2   CoreFoundation                      0x03d9e32a +[NSException raise:format:arguments:] + 138
	3   Foundation                          0x01e74322 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 118
       ...
```

**Comment by alfonsogarciacaro** [comment]
Note this seems to be the same error [automatically reported](https://github.com/toggl/mobile/issues?q=is%3Aopen+is%3Aissue+label%3A%22automatic+report%22) many times by TestFairy.

**Comment by alfonsogarciacaro** [comment]
Related report: https://toggl.desk.com/agent/case/291830

**Comment by amulware** [comment]
From googling the error it seems that this may have something to do with setting the date while the date picker is not visible/focused. This does not explain why it works with time entries created with the new version though. Still investigating what difference between the two could cause this.

**Comment by alfonsogarciacaro** [comment]
At first sight, it looked to me that the problem affected only to old entries but I didn't check very exhaustively. It may be that I'm wrong.

**Comment by amulware** [comment]
I will investigate that more than, however it does seem the same way to me.

**Comment by amulware** [comment]
Alright. I just found some old time entries that do not crash, so it must(?) be something else.

**Comment by amulware** [comment]
New info: time entries with a start-time past 3pm cause the crash. others do not (as far as I can tell).

**Comment by amulware** [comment]
Closed by #1611.
